packages/typespec-ts/test/modularUnit/scenarios/samples/client/multipleClient.md

```diff
@@ -0,0 +1,98 @@
+# Should generate samples for enabled hierarchy client
+
+Sample generation should handle hierarchy client successfully.
+
+## TypeSpec
+
+This is tsp definition.
+
+```tsp
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-client-generator-core";
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.ClientGenerator.Core;
+
+@service({
+  title: "Demo Service",
+})
+@versioned(Versions)
+@useAuth(ApiKeyAuth<ApiKeyLocation.header, "api-key">)
+namespace DemoService;
+
+enum Versions {
+  /** Version 2021-10-01-preview */
+  `2021-10-01-preview`,
+}
+@route("b")
+namespace Sub {
+  @route("foo")
+  op foo(): void;
+  @route("bar")
+  op bar(): void;
+}
+
+@client({
+  name: "AClient",
+  service: DemoService,
+})
+interface ClientA {
+  foo is Sub.foo;
+}
+
+
+@client({
+  name: "BClient",
```

Comment:
Can you add a case for BClient? and make the client constructor is different?

Reply:
you mean different client would have different constructor, i don't think this is supported yet. we could achive this once https://github.com/Azure/typespec-azure/pull/1398 is released.